src/main/java/frc/robot/subsystems/TCS3200_ColorSensor.java

```diff
@@ -0,0 +1,200 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.Counter;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DigitalOutput;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class TCS3200_ColorSensor extends SubsystemBase {
+  // Output managment
+  private Counter m_counter;
+
+  // Configurable registers
+  private DigitalOutput m_s0;
+  private DigitalOutput m_s1;
+  private DigitalOutput m_s2;
+  private DigitalOutput m_s3;
+
+  private FreqScaling m_freqScaling;
+  private ColorSelect m_currColor;
+
+  private double m_lastTimeStamp;
+
+  public enum FreqScaling {
+    POWER_DOWN(0),
+    PERCENT2(1),
+    PERCENT20(2),
+    PERCENT100(3);
+
+    public final int val;
+
+    FreqScaling(int i) {
+      val = i;
+    }
+  }
+
+  public enum ColorSelect {
+    RED(0),
+    GREEN(1),
+    BLUE(2),
+    CLEAR(3);
+
+    public final int val;
+
+    ColorSelect(int i) {
+      val = i;
+    }
+  }
+
+  private double m_red;
+  private double m_green;
+  private double m_blue;
+  private double m_clear;
+
+  public TCS3200_ColorSensor(int out, int s0, int s1, int s2, int s3, ColorSelect startColor) {
+    m_counter = new Counter(new DigitalInput(out));
+
+    // To save DIO ports, let user not set some
+    if (s0 != -1) {
+      m_s0 = new DigitalOutput(s0);
+    }
+
+    if (s1 != -1) {
+      m_s1 = new DigitalOutput(s1);
+    }
+
+    if (s2 != -1) {
+      m_s2 = new DigitalOutput(s2);
+    }
+
+    if (s3 != -1) {
+      m_s3 = new DigitalOutput(s3);
+    }
```

Comment:
The if statements cannot be the most efficient way to structure this, we also want default values for all of them

Reply:
resolved in e48b7db5abfdf7b56b1083fc0df72ed9ebe5214f